### ISSUE TITLE
Smoothing the earnings distribution

### ISSUE THREAD

**Issue by jdebacker** [PSLmodels/InverseOptimalTax#18]
@john-p-ryan has done good work on how to allow one to use a KDE to smooth the bottom part of the earnings distribution and be able to pair that with a Pareto distribution for the upper tail.

We should implement this as a viable method in the `compute_income_dist` method.  A more consistent estimate of the upper part of the distribution (which has few observations) will help to get more informative estimates of the social welfare weights.

**Comment by jdebacker** [comment]
We now use a parametric distribution, which is assumed to be log-normal.  

I think a reasonable extension is to fit the upper tail with a Pareto distribution.

**Comment by jdebacker** [comment]
@john-p-ryan: Per our conversation yesterday about fitting a joint log-normal and Pareto distribution, here's a picture that replicates Figure 2 from [Saez (2001)](https://eml.berkeley.edu/~saez/derive.pdf) with the CPS data on wages and salaries we used, grown out to 2023 values:

![Saez2001_Fig2_2023_data](https://github.com/PSLmodels/InverseOptimalTax/assets/10715924/00f9e482-2a52-4b00-96f6-6504aa39ba40)

And the figure from Saez:

![Screenshot 2024-01-31 at 8 59 15 AM](https://github.com/PSLmodels/InverseOptimalTax/assets/10715924/5cdf2507-f4e3-4bb6-9605-3e9baa1d8426)

It would appear to me that our estimates of the Pareto parameter for the tail of the distribution would be an $\alpha\approx 1.9$ and that our cutoff value for the tail would be in the $150,00-$200,000 range.

Code to replicate:
```
iot_2023 = iot_user.iot_comparison(
        policies=[{}],
        baseline_policies=[None],
        labels=["2023 Law"],
        years=[2023],
        data="CPS",
    )
iot_2023.SaezFig2()
```

**Comment by jdebacker** [comment]
Resolved with PR #26
